host/sgx/create.c

```diff
@@ -1005,8 +1096,19 @@ oe_result_t oe_create_enclave(
     enclave->ocalls = (const oe_ocall_func_t*)ocall_table;
     enclave->num_ocalls = ocall_count;
 
-    /* Invoke enclave initialization. */
-    OE_CHECK(_initialize_enclave(enclave));
+/* Invoke enclave initialization. */
+#ifdef OE_WITH_EXPERIMENTAL_EEID
+    if (context.eeid_setting)
+    {
+        oe_enclave_with_config_t enclave_with_config = {0};
+        enclave_with_config.enclave = enclave;
+        enclave_with_config.config.data = (uint8_t*)context.eeid_setting->data;
+        enclave_with_config.config.size = context.eeid_setting->data_size;
+        OE_CHECK(_initialize_enclave_with_config(&enclave_with_config));
```

Comment:
Instead of using a "config_data" specific `_initialize_enclave` call, can follow the design of the switchless call implementaiton, and use a dedicated `oe_load_config_data` ECALL, invoked within `_config_enclave` (line 1131), if the `settings` indicate the presence of config_data. This way, the enclave initialization code is not affected and `oe_load_config_data` ECALL happens after all necessary initalizations are done. Future changes to the enclave intialization design is much less likely to cause any compatibility issue with `oe_load_config_data` flow. If the developer of an enclave supporting loading additional content post initalization (EEID enclave or future SGX KSS-based implementation) does not provide config_data through `settings`, the devloper can implement their own ECALL to load additional content into the enclave, with thier own definition of config_id/config_svn (config_id is not necceary a hash of config_data). Note that not loading any additional data is a legit usage of Enclaves that support loading additional data.

Reply:
Outdated. Regardless of what we do, the developer can always add their own ecall and overwrite (or just ignore) the default config data. I won't touch this until the design discussion is finished and the new ecall is signed off on.